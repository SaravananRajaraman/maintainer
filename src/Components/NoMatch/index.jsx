import React from 'react';
import styled from 'styled-components';
import Typography from '@material-ui/core/Typography';

const StyledBox = styled('div')({
    display : 'grid',
    alignItems : 'center',
    justifyItems : 'center',
    height : '100vh'
})

export default function NoMatch() {
  return (
    <StyledBox data-test='NoMatch-Component'>
        <div>
            <Typography variant='h1' data-test='Error'>{'404'}</Typography>
            <Typography variant='h6' data-test='Description'>{'Page not found!'}</Typography>
        </div>        
    </StyledBox>
  );
};