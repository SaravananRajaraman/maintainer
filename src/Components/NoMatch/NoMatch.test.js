import React from 'react';
import {shallow} from 'enzyme';
import {findTestAttr} from './../../Utils';
import NoMatch from './index';
import toJson from 'enzyme-to-json';

const setUp =()=>{
    const component = shallow(<NoMatch/>);
    return component;
}

describe('No Match Component', ()=>{


    let component;
    beforeEach(()=>{
        component = setUp();
    })

    it('Should render the component with no error', ()=>{        
        let wrapper = findTestAttr(component,'NoMatch-Component');
        expect(wrapper.length).toBe(1);
    });

    it('Should render error', ()=>{        
        let wrapper = findTestAttr(component,'Error');
        expect(wrapper.length).toBe(1);
    });

    it('Should render error description', ()=>{        
        let wrapper = findTestAttr(component,'Description');
        expect(wrapper.length).toBe(1);
    });

    it('Should match the snapShot', ()=>{                
        expect(toJson(component)).toMatchSnapshot();
    });

});
