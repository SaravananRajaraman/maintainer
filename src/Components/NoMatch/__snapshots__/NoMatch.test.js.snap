// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No Match Component Should match the snapShot 1`] = `
<styled.div
  data-test="NoMatch-Component"
>
  <div>
    <WithStyles(ForwardRef(Typography))
      data-test="Error"
      variant="h1"
    >
      404
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      data-test="Description"
      variant="h6"
    >
      Page not found!
    </WithStyles(ForwardRef(Typography))>
  </div>
</styled.div>
`;
